src/enzyme_ad/jax/Passes/PropagateConstantBound.cpp

```diff
@@ -34,15 +34,32 @@ struct PropagateConstantBoundsPass
     : public enzyme::impl::PropagateConstantBoundsPassBase<
           PropagateConstantBoundsPass> {
 
+  // If we know that this range is constant, we attach the LLVM range attribute
+  // to the target. If the target already has a range, we update it by taking
+  // the maximum between the current value and the old value to be conservative.
   static void attachConstantRangeIfConstant(MLIRContext *ctx,
                                             Operation *maybeCst,
                                             Operation *target) {
     APInt intValue;
-    if (matchPattern(maybeCst, m_ConstantInt(&intValue)))
-      target->setAttr("range", LLVM::ConstantRangeAttr::get(
-                                   ctx, 32, 0, intValue.getSExtValue()));
+    if (matchPattern(maybeCst, m_ConstantInt(&intValue))) {
```

Comment:
I don't think this is the best way to do things.

If, for example, there is already range metadata (that is weaker), we won't do the update.

I think the thing to do here is to
1) get all kernelcall ops
2) then get all called functions
3) for each called function
    a) loop over all callers, getting the max over all ranges/etc
    b) set the derived values/aliasing/etc

Reply:
I did not understand your previous comment then. What do you mean by "weaker range"? Can you provide an example?

Reply:
Yeah supposed that the existing kernel code before this pass has a range of 1-100000, but all the callers have a range 1-100. We would be taking the max over the previous setting and all the actual callers